src/main/java/com/softserve/security/user/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package com.softserve.security.user;
+
+import com.softserve.model.User;
+import com.softserve.repository.IUserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    @Autowired
+    IUserRepository userRepository;
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User Not Found with username: " + username));
```

Comment:
is this exception message passed to the UI?

If your answer is yes, I would advise not doing it, usually, when talking about security during login, we just respond to the user/client saying "username and/or password is incorrect", meaning, we do not facilitate to a potential attacker which of the fields is incorrect or does not exist.

Reply:
Ow, nope, sorry, this exceptions is not passed to UI, this error onlye can be see from log files.